objectherkenning_openbare_ruimte/convert_dataset_pipeline/source/equirectangular_to_cubemap_converter.py

```diff
@@ -0,0 +1,752 @@
+
+logger = logging.getLogger(__name__)
+
+
+class EquirectangularToCubemapConverter:
+    """
+    - Convert an equirectangular image to a set of six cubic faces using py360convert library
+    (https://github.com/imandrealombardo/py360convert)
+    - Convert YOLO format annotations to the corresponding cubic faces.
+    - Visualize the annotations on both the equirectangular and cubic images.
+    """
+
+    def __init__(self, input_path: str, output_path: str, face_width: int):
+        """
+        Parameters
+        ----------
+        input_path : str
+            Path to the equirectangular image folder.
+        output_path : str
+            Path to the folder where to store the folders with the six cubic faces for each image.
+        """
+        self.input_path = input_path
+        self.output_path = output_path
+        self.face_width = face_width
+
+    @staticmethod
+    def _convert_face_idx_to_name(face_idx: int) -> str:
+        """
+        Convert face index to face name.
+
+        The mapping is:
+        0 -> front, 1 -> right, 2 -> back,
+        3 -> left, 4 -> top, 5 -> bottom.
+
+        Parameters
+        ----------
+        face_idx : int
+            Index of the face.
+
+        Returns
+        -------
+        str
+            Name of the face.
+        """
+        face_names = ["front", "right", "back", "left", "top", "bottom"]
+        return face_names[face_idx]
+
+    @staticmethod
+    def _convert_yolo_to_corners(yolo_annotation: str, P_w: int, P_h: int) -> Tuple[
+        Tuple[float, float],
+        Tuple[float, float],
+        Tuple[float, float],
+        Tuple[float, float],
+    ]:
+        """
+        Convert YOLO annotation with normalized values to absolute corner coordinates.
+
+        Parameters
+        ----------
+        yolo_annotation : str
+            YOLO annotation string in the format:
+            "<class_id> <x_center_norm> <y_center_norm> <w_norm> <h_norm>".
+        P_w : int
+            Width of the image.
+        P_h : int
+            Height of the image.
+
+        Returns
+        -------
+        tuple
+            A tuple containing the top-left, top-right, bottom-left,
+            and bottom-right corner coordinates.
+        """
+
+        _, x_center_norm, y_center_norm, w_norm, h_norm = map(
+            float, yolo_annotation.split()
+        )
+
+        x_center_abs = x_center_norm * P_w
+        y_center_abs = y_center_norm * P_h
+        w_abs = w_norm * P_w
+        h_abs = h_norm * P_h
+
+        x_min = x_center_abs - (w_abs / 2)
+        y_min = y_center_abs - (h_abs / 2)
+        x_max = x_center_abs + (w_abs / 2)
+        y_max = y_center_abs + (h_abs / 2)
+
+        top_left = (x_min, y_min)
+        top_right = (x_max, y_min)
+        bottom_left = (x_min, y_max)
+        bottom_right = (x_max, y_max)
+
+        return top_left, top_right, bottom_left, bottom_right
+
+    @staticmethod
+    def _convert_corners_to_yolo(
+        yolo_class: str,
```

Comment:
`yolo_class` is just passed through, might be more "clean" to concatenate in the calling method and only do the coordinate logic here.